include/camera.h

```diff
@@ -0,0 +1,95 @@
+#ifndef POKEPLATINUM_CAMERA_H
+#define POKEPLATINUM_CAMERA_H
+
+#include "overlay115/camera_angle.h"
+#include "constants/heap.h"
+
+#include <nitro/fx/fx.h>
+
+#define CAMERA_DELAY_NONE   0
+#define CAMERA_DELAY_X      1
+#define CAMERA_DELAY_Y      2
+#define CAMERA_DELAY_Z      4
+
+enum CameraProjection {
+    CAMERA_PROJECTION_PERSPECTIVE = 0,
+    CAMERA_PROJECTION_ORTHOGRAPHIC,
+};
+
+typedef struct CameraLookAt {
+    VecFx32 position;
+    VecFx32 target;
+    VecFx32 up;
```

Comment:
**question:** How does this field affect the camera? In my mind, the camera is at a position and pointed at a target, which defines the line of view between those two points.

Reply:
Well just a position and a target defines the X and Y rotation of how the camera is looking at the target, but it doesn't define a Z rotation. Given just a position and target the camera could be looking at the target upside down, sideways, etc. That's why you define an up vector.

Technically you can even use the up vector to simulate roll. E.g. If you add a `theta` field to the camera and put this inside `Camera_ComputeViewMatrix`:
```c
// Simulate "camera roll"
sActiveCamera->theta = (sActiveCamera->theta + 100) % 0x10000;
sActiveCamera->lookAt.up.x = FX_Mul(FX_SinIdx(sActiveCamera->theta), -FX32_ONE);
sActiveCamera->lookAt.up.y = FX_CosIdx(sActiveCamera->theta);
sActiveCamera->lookAt.up.z = 0;
```
You get the following:

https://github.com/pret/pokeplatinum/assets/60443001/755b2f7f-8814-42a6-8c52-a5c25a75e42c

Or if you simply define the up vector as `<0, -1, 0>` your game will simply be upside down:

https://github.com/pret/pokeplatinum/assets/60443001/e48ac462-bcb8-4f9c-8e03-8dcc819926ad

Reply:
I assume then that this is how the Distortion World... "works," for lack of a better term. Neat! Very interesting. Thanks for providing the video evidence. :grin: